### PR TITLE
feat(ui) Update design of issue searchbar

### PR DESCRIPTION
**Pull request by markstory**
Update the visuals of the issue searchbar to match the latest mocks. I've moved the various buttons into a container div that exists outside the form to address the submit issues that folks are having.

I've duplicated a bunch of display logic in SmartSearchBar. I was aiming to make the code easy to clean-up later as I've found that larger conditionals are easier to remove than smaller ones.

Screen Shot 2019-04-16 at 12 05 31 PM

Fixes SEN-474

### CODE REVIEW

File: tests/js/spec/components/smartSearchBar.spec.jsx [hunk 1]
```diff
@@ -260,6 +277,28 @@ describe('SmartSearchBar', function() {
...
+          orgId="123"
+          projectId="456"
+          query="is:unresolved"
+          supportedTags={supportedTags}
+          hasPinnedSearch
+        />,
+        options
+      );
+
+      wrapper.find('form').simulate('submit', {
```
Comment (billyvg) [inline tests/js/spec/components/smartSearchBar.spec.jsx]: Would simulating an enter keypress fire a form submit? Can we test for that instead?

File: tests/js/spec/views/stream/searchBar.spec.jsx [hunk 2]
```diff
@@ -275,7 +275,7 @@ describe('SearchBar', function() {
         pinnedSearch: {id: '1', query: 'url:"fu" '},
       };
       const searchBar = mount(<SearchBar {...props} />, routerContext);
-      searchBar.find('PinButton').simulate('click');
+      searchBar.find('button[data-test-id="pin-search"]').simulate('click');
```
Comment (billyvg) [inline tests/js/spec/views/stream/searchBar.spec.jsx]: We should consider requiring aria-labels for all Buttons for accessibility and use it as selectors for tests

**Review by billyvg — COMMENTED** [review]
Is the search bar going to look different between issues search and events search?

We will eventually want the `ButtonBar` to be configurable because I could see Events search having a pinned search and/or saved searches, but I think this is good for now.

File: tests/js/spec/views/stream/searchBar.spec.jsx (hunk 2 above)
Comment (markstory) [reply tests/js/spec/views/stream/searchBar.spec.jsx]: I can add aria labels to these buttons.

File: tests/js/spec/components/smartSearchBar.spec.jsx (hunk 1 above)
Comment (markstory) [reply tests/js/spec/components/smartSearchBar.spec.jsx]: Sure thing. I was basing this off the existing tests, but I'll change it around.

**Comment by markstory** [comment]
> Is the search bar going to look different between issues search and events search?

Right now I've not changed how the search bar looks for event search.

**Review by billyvg — APPROVED** [review]

**Pull request merged**